probe/endpoint/ebpf.go

```diff
@@ -287,10 +310,33 @@ func (t *EbpfTracker) isDead() bool {
 
 func (t *EbpfTracker) stop() {
 	t.Lock()
+	defer t.Unlock()
+
```

Comment:
Actually, it fails if I let the Unlock as it was before:

```
<probe> WARN: 2017/07/24 13:45:09.816261 ebpf tracker stopped as requested by user
<probe> WARN: 2017/07/24 13:45:10.304710 ebpf tracker died, trying to restart it
<probe> WARN: 2017/07/24 13:45:10.312468 could not restart ebpf tracker, falling back to proc scanning: cannot write "p:ptcp_v4_connect tcp_v4_connect\n" to kprobe_events: write /sys/kernel/debug/tracing/kprobe_events: device or resource busy
```

The stop is done in the event handler thread, when I receive an event with a wrong timestamp. And the restart is done in the publisher thread when it notices that `isDead == true`. I think that the publisher thread starts the restart before the tracer is fully stopped, so the kprobe is not completely removed yet...